File: Medium/2026-04-08-3653-XOR-After-Range-Multiplication-Queries-I/solution.js
var xorQueries = function(nums, queries) {
    const MOD = 1e9 + 7;
    const n = nums.length;

    for (const [l, r, k, v] of queries) {
        for (let idx = l; idx <= r; idx += k) {
            nums[idx] = (nums[idx] * v) % MOD;
        }
    }

    return nums.reduce((acc, num) => acc ^ num, 0);
};

File: Medium/2026-04-08-3653-XOR-After-Range-Multiplication-Queries-I/solution.java
class Solution {
    public int xorAfterQueries(int[] nums, int[][] queries) {
        int MOD = 1000000007;
        for (int[] query : queries) {
            int l = query[0], r = query[1], k = query[2], v = query[3];
            for (int idx = l; idx <= r; idx += k) {
                nums[idx] = (int)((long)nums[idx] * v % MOD);
            }
        }
        
        int result = 0;
        for (int num : nums) {
            result ^= num;
        }
        
        return result;
    }
}